test/jdk/java/util/zip/ZipFile/ZipFileDuplicateEntryTest.java

```diff
@@ -0,0 +1,585 @@
+        if (DEBUG) {
+            System.out.printf("URL=%s%n", url);
+        }
+        JarURLConnection con = (JarURLConnection) url.openConnection();
+        con.connect();
+        JarEntry je = con.getJarEntry();
+        try (JarFile jarFile = con.getJarFile()) {
+            assertNotNull(je);
+            assertNotNull(jarFile);
+            assertNull(con.getAttributes());
+            assertNull(con.getMainAttributes());
+            assertNull(con.getManifest());
+            assertEquals(je.getName(), entry.name);
+            assertEquals(con.getEntryName(), entry.name);
+            assertEquals(je.getMethod(), entry.method);
+            assertEquals(con.getJarFileURL(), TEST_JAR.toUri().toURL());
+            if (DEBUG) {
+                System.out.printf("   getEntryName: %s,  getJarFileURL:%s%n",
+                        con.getEntryName(), con.getJarFileURL());
+                System.out.printf("   Jar Entry= %s, size= %s%n", je.getName(), je.getSize());
+            }
+
+            try (InputStream is = jarFile.getInputStream(je)) {
+                byte[] bytes = is.readAllBytes();
+                if (DEBUG) {
+                    System.out.printf("   Bytes read:%s%n", new String(bytes));
+                }
+                assertEquals(bytes, entry.bytes,
+                        String.format("Expected payload: %s", new String(entry.bytes)));
+            }
+        }
+    }
+
+    /**
+     * Verify that JarFile::stream returns all entries including
+     * a file and directory entry with the same name
+     *
+     * @throws IOException If an error occurs
+     */
+    @Test
+    public void JarFileStreamTest() throws IOException {
+        System.out.printf("%n%n**** JarFileStreamTest ***%n");
+        try (JarFile jf = new JarFile(TEST_JAR.toFile())) {
+            List<? extends JarEntry> entries = jf.stream().collect(Collectors.toList());
+            assertEquals(EXPECTED_ENTRIES.size(), jf.size());
+            for (JarEntry e : entries) {
+                assertTrue(EXPECTED_ENTRIES.contains(e.getName()));
+            }
+        }
+    }
+
+    /**
+     * Method used to read  the bytes from an InputStream.  This method is
+     * here so that the test could be backported to JDK 8 if needed as
+     * InputStream::readAllBytes() does not exist
+     *
+     * @param is The InputStream to read from
+     * @return The byte array representing bytes read from the InputStream
+     * @throws IOException If an error occurs
+     */
+    public static byte[] readAllBytes(InputStream is) throws IOException {
+        byte[] data = new byte[MAX_BUFFER_SIZE];
+        ByteArrayOutputStream buffer = new ByteArrayOutputStream();
+        int len;
+        while ((len = is.read(data, 0, data.length)) != -1) {
+            buffer.write(data, 0, len);
+        }
+        buffer.flush();
+        return buffer.toByteArray();
+    }
+
+    /**
```

Comment:
Left-over?

Reply:
Thank you for the quick review Claes!

I was going to remove this method but chose to leave it in to make it a bit more straight forward to port to JDK 8 (which I have run the test on as well).  No strong preferences, just figured leaving it there might make it slightly easier for a back port.